### PR TITLE
feat: replace fast-glob with tinyglobby

### PR DESCRIPTION
**Pull request by SuperchupuDev**
relevant: https://github.com/unocss/unocss/pull/3722#issuecomment-2053769072

[`tinyglobby`](https://github.com/es-tooling/module-replacements/blob/main/docs/modules/glob.md#tinyglobby) is meant to be a drop-in replacement to `globby` and `fast-glob`, and it only uses two subdependencies - `fdir` and `picomatch`. it's already being used in some somewhat big packages like [vite-plugin-pwa](https://github.com/vite-pwa/vite-plugin-pwa/pull/733) with no user reports of breaking behavior.

`expandDirectories` is explicitly disabled for better `fast-glob` compatibility

### CODE REVIEW

File: packages/postcss/src/esm.ts [hunk 1]
```diff
@@ -105,12 +104,12 @@ export function createPlugin(options: UnoPostcssPluginOptions) {
     const globs = uno.config.content?.filesystem ?? defaultFilesystemGlobs
     const plainContent = uno.config.content?.inline ?? []
 
-    const entries = await fg(isScanTarget ? globs : from, {
+    const entries = await glob(isScanTarget ? globs : [from], {
```
Comment (userquin) [inline packages/postcss/src/esm.ts:107]: cannot add `then` and find stats using `node:fs/promises`, we should add small cache to avoid process unchanged files

Comment (SuperchupuDev) [reply packages/postcss/src/esm.ts:107]: i'm not sure i understand, do you mean using `node:fs/promises`' `stats` to use `fileMap` the same way it was used before?

Comment (userquin) [reply packages/postcss/src/esm.ts:107]: yes, if the logic was wrong just fix it

**Comment by SuperchupuDev** [comment]
tests should pass now, fixed an edge case around empty patterns

File: packages/postcss/src/esm.ts [hunk 2]
```diff
@@ -135,14 +135,16 @@ export function createPlugin(options: UnoPostcssPluginOptions) {
 
     for (let i = 0; i < entries.length; i += BATCH_SIZE) {
       const batch = entries.slice(i, i + BATCH_SIZE)
-      promises.push(...batch.map(async ({ path: file, mtimeMs }) => {
+      promises.push(...batch.map(async (file) => {
         result.messages.push({
```
Comment (userquin) [inline packages/postcss/src/esm.ts:139]: is this required? I mean, shouldn't be added only if the file has changed?

File: packages/postcss/src/esm.ts [hunk 3]
```diff
@@ -135,14 +135,16 @@ export function createPlugin(options: UnoPostcssPluginOptions) {
...
           type: 'dependency',
           plugin: directiveMap.unocss,
           file: normalize(file),
           parent: from,
         })
 
+        const { mtimeMs } = await stat(file)
+
         if (fileMap.has(file) && mtimeMs <= fileMap.get(file))
           return
```
Comment (userquin) [inline packages/postcss/src/esm.ts:150]: can you remove the `else`? We've return if the file is not changed

File: packages/postcss/src/esm.ts (hunk 2 above)
Comment (SuperchupuDev) [reply packages/postcss/src/esm.ts:139]: hmm, would moving `result.messages.push` after the `fileMap` check be a better option?

Comment (userquin) [reply packages/postcss/src/esm.ts:139]: no idea, I'm just asking...

Comment (SuperchupuDev) [reply packages/postcss/src/esm.ts:139]: it makes sense to move it (at least to me), and tests seem to pass, so i'll push it
